### ISSUE TITLE
Installing HoTT results in many 'ambiguous-file-name' messages for .cmx plugins

### ISSUE THREAD

**Issue by MSoegtropIMC** [HoTT/HoTT#1064]
Dear HoTT Team,

after addign HoTT to the Windows Installer, I get quite a few `ambiguous-file-name` warnings, e.g. when doing `Require Import Reals`. This results in:
```
r_syntax_plugin.cmxs has been found in
[ D:\bin\coq-810-64-hott-3\lib\coq\user-contrib\hott\coq\plugins\syntax ;
  D:\bin\coq-810-64-hott-3\lib\coq\plugins\syntax ];
loading D:\bin\coq-810-64-hott-3\lib\coq\user-contrib\hott\coq\plugins\syntax\r_syntax_plugin.cmxs
[ambiguous-file-name,filesystem]
[Loading ML file r_syntax_plugin.cmxs ... done]
newring_plugin.cmxs has been found in
[ D:\bin\coq-810-64-hott-3\lib\coq\user-contrib\hott\coq\plugins\setoid_ring ;
  D:\bin\coq-810-64-hott-3\lib\coq\plugins\setoid_ring ];
loading D:\bin\coq-810-64-hott-3\lib\coq\user-contrib\hott\coq\plugins\setoid_ring\newring_plugin.cmxs
[ambiguous-file-name,filesystem]
[Loading ML file newring_plugin.cmxs ... done]
omega_plugin.cmxs has been found in
[ D:\bin\coq-810-64-hott-3\lib\coq\user-contrib\hott\coq\plugins\omega ;
  D:\bin\coq-810-64-hott-3\lib\coq\plugins\omega ];
loading D:\bin\coq-810-64-hott-3\lib\coq\user-contrib\hott\coq\plugins\omega\omega_plugin.cmxs
[ambiguous-file-name,filesystem]
[Loading ML file omega_plugin.cmxs ... done]
micromega_plugin.cmxs has been found in
[ D:\bin\coq-810-64-hott-3\lib\coq\user-contrib\hott\coq\plugins\micromega ;
  D:\bin\coq-810-64-hott-3\lib\coq\plugins\micromega ];
loading D:\bin\coq-810-64-hott-3\lib\coq\user-contrib\hott\coq\plugins\micromega\micromega_plugin.cmxs
[ambiguous-file-name,filesystem]
[Loading ML file micromega_plugin.cmxs ... done]
```
A few questions on this:
- does this also happen on Linux (if `make install` for HoTT is called)
- why are these plugins recompiled? Is it dangerous if it loads the HoTT version when I am not doing HoTT?
- Do you have suggestions on how this could be solved?

**Comment by MSoegtropIMC** [comment]
P.S.: I had to suspend the PRs to include HoTT in the Coq release Windows installers cause of this. See
https://github.com/coq/coq/pull/10588
https://github.com/coq/coq/pull/10589

**Comment by SkySkimmer** [comment]
My opam install has hott stuff in `.opam/$switch/share/hott` and `hoq*` wrappers (without `hoqide` for some reason) in `.opam/$switch/bin`
Notably `.opam/$switch/share/hott/coq` contains the compilation products (vo etc) in `theories/` for the stdlib stubs, and symbolic links to the coq install for `dev` `kernel` `library` `plugins` and `stm` (why those? I don't know) (replacing by copies works fine)

Since it's not in Coq's default search paths (user-contrib) there are no warnings.

Indeed installing in user-contrib seems wrong, to use hott you need to pass -coqlib and so on, so it shouldn't be treated as a normal package.

**Comment by MSoegtropIMC** [comment]
Good point - I should put the HoTT stuff somewhere else. I will do some experiments and see if share/hott does work (I am not 100% sure because at least on Windows it also scans share for .vo files, but maybe they are treated differently).

**Comment by JasonGross** [comment]
> (why those? I don't know) (replacing by copies works fine)

Those were the ones that I had to add to get coqtop not to barf when I passed it a strange coqlib, IIRC.  Copies to work fine, and `./configure` is set up to make copies on platforms where symlinks don't reliably work (such as Windows).

> without hoqide for some reason

It's possible that something went wrong with the coqide detection, and it only installs hoqide if you have coqide installed?
